### PR TITLE
[17.0][MIG] sale_order_general_discount: 17.0

### PR DESCRIPTION
**Pull request by miguel-S73**
Standard migration from 16.0 to 17.0 and I have added  `cherry-pick` https://github.com/OCA/sale-workflow/pull/2524 and this suggestion https://github.com/OCA/server-backend/pull/221#pullrequestreview-1664274601 too.

### CODE REVIEW

**Comment by gbtechnology** [comment]
@miguel-S73 some tests failed, do you need some help or you are already working on this?

**Comment by miguel-S73** [comment]
@gbtechnology now I have fixed red flags

**Comment by gbtechnology** [comment]
@miguel-S73 cool!
Who is suppose to approve/merge this?

**Comment by miguel-S73** [comment]
@sergio-teruel @pedrobaeza @ivantodorovich @rven 
Anyone can review my PR?

**Comment by pedrobaeza** [comment]
Please say to one of your colleagues to review it and I'll review it too.

/ocabot migration sale_order_general_discount

File: sale_order_general_discount/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    general_discount_apply = fields.Boolean(
+        string="Apply general discount",
+        default=True,
+        required=True,
```
Comment (pedrobaeza) [inline sale_order_general_discount/models/product_product.py]: `required` in a boolean is useless.

File: sale_order_general_discount/models/product_product.py [hunk 2]
```diff
@@ -6,10 +6,8 @@
 class ProductProduct(models.Model):
     _inherit = "product.product"
 
-    general_discount_apply = fields.Boolean(
-        string="Apply general discount",
-        default=True,
-        required=True,
-        help="If this checkbox is ticked, it means changing general discount on sale order "
-        "will impact sale order lines with this related product.",
+    not_general_discount_apply = fields.Boolean(
```
Comment (pedrobaeza) [inline sale_order_general_discount/models/product_product.py]: It's not convenient to name negative fields. Better something like `bypass_general_discount`.

File: sale_order_general_discount/models/sale_order_line.py [hunk 3]
```diff
@@ -1,11 +1,33 @@
...
     _inherit = "sale.order.line"
 
+    discount = fields.Float(
+        compute="_compute_discount",
+        store=True,
+        readonly=False,
+    )
+
+    @api.model_create_multi
+    def create(self, vals_list):
```
Comment (pedrobaeza) [inline sale_order_general_discount/models/sale_order_line.py]: I don't like this override, and doing the field computed writable shouldn't be needed.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: sale_order_general_discount/models/product_product.py (hunk 1 above)
Comment (miguel-S73) [reply sale_order_general_discount/models/product_product.py]: I removed this attribute in my last commit. My intention was to do a `cherry-pick` and then apply my changes.

File: sale_order_general_discount/models/product_product.py (hunk 2 above)
Comment (miguel-S73) [reply sale_order_general_discount/models/product_product.py]: Now I have  applied this suggested change

File: sale_order_general_discount/models/sale_order_line.py (hunk 3 above)
Comment (miguel-S73) [reply sale_order_general_discount/models/sale_order_line.py]: I have check funcionality after I remove this override and work fine.

**Review by pedrobaeza — APPROVED** [review]

File: sale_order_general_discount/models/product_product.py [hunk 4]
```diff
@@ -6,10 +6,8 @@
 class ProductProduct(models.Model):
     _inherit = "product.product"
 
-    general_discount_apply = fields.Boolean(
-        string="Apply general discount",
-        default=True,
-        required=True,
-        help="If this checkbox is ticked, it means changing general discount on sale order "
-        "will impact sale order lines with this related product.",
+    bypass_general_discount = fields.Boolean(
```
Comment (rousseldenis) [inline sale_order_general_discount/models/product_product.py:9]: Shouldn't that change need migration script ?

Comment (miguel-S73) [reply sale_order_general_discount/models/product_product.py:9]: @rousseldenis 
I don't think it's needed.
I made this change from a `cherry-pick` from another PR and  the `general_discount_apply` field does not exist in previous version of this module.

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**